java/java.editor.base/test/unit/src/org/netbeans/modules/java/editor/base/semantic/TestBase.java

```diff
@@ -197,8 +201,74 @@ public void run(CompilationController parameter) {
         }
     }
     
-    protected ColoringAttributes getColoringAttribute() {
-        return ColoringAttributes.UNUSED;
+    protected void performTest(String fileName, String code, Performer performer, String... expected) throws Exception {
+        performTest(fileName, code, performer, false, expected);
+    }
+
+    protected void performTest(String fileName, String code, Performer performer, boolean doCompileRecursively, String... expected) throws Exception {
+        SourceUtilsTestUtil.prepareTest(new String[] {"org/netbeans/modules/java/editor/resources/layer.xml"}, new Object[] {new MIMEResolverImpl()});
+
+	FileObject scratch = SourceUtilsTestUtil.makeScratchDir(this);
+	FileObject cache   = scratch.createFolder("cache");
+
+        File wd         = getWorkDir();
+        File testSource = new File(wd, "test/" + fileName + ".java");
+
+        testSource.getParentFile().mkdirs();
+        TestUtilities.copyStringToFile(testSource, code);
+
+        testSourceFO = FileUtil.toFileObject(testSource);
+
+        assertNotNull(testSourceFO);
+
+        if (sourceLevel != null) {
+            SourceUtilsTestUtil.setSourceLevel(testSourceFO, sourceLevel);
+        }
+
+        File testBuildTo = new File(wd, "test-build");
+
+        testBuildTo.mkdirs();
+
+        FileObject srcRoot = FileUtil.toFileObject(testSource.getParentFile());
+        SourceUtilsTestUtil.prepareTest(srcRoot,FileUtil.toFileObject(testBuildTo), cache);
+
```

Comment:
Jackpot:
warning: Redundant type arguments in new expression (use diamond operator instead).